### ISSUE TITLE
`anemic-getter` lint incorrectly flags explicit receiver as a redundant rename

### ISSUE THREAD

**Issue by maxonfjvipon** [objectionary/lints#1300]
In `eo-runtime` we are making the receiver explicit: every formation head declares `^`, so the receiver becomes an ordinary void filled by dispatch rather than arriving through `AtWithRho` at load time. That turns eighty-four existing bindings into lint failures.

The idiom is a formation giving its receiver a name:

```
[^ x] > xor
  or. > @
    b.and x.not
    b.not.and x
  ^ > b
```

Once `ρ` is a declared void, `^ > b` reads as a getter renaming an attribute, and the lint says `The object "b" is a redundant getter, it just renames "ρ" without adding anything, use "ρ" directly instead`.

For a flat body the advice is fine. It stops being fine as soon as an inner formation uses the name, because the replacement is not `^` but a chain. In `chunk/to-output.eo` the body reads `^.^.^.allocated`, and dropping the binding makes that `^.^.^.^`.

Those two are equal only for the nesting that exists today. `allocated` names one thing no matter where it is written; `^.^.^.^` names whatever happens to be four hops up, so re-indenting a block or introducing a wrapper silently retargets it. We tried the rewrite and CI found the breakage — a `StackOverflowError` in the README snippets and errors across `map`, `while`, `range` and `input` — which is a fair summary of how easy it is to get the hop count wrong.

So the binding does add something: a name that survives nesting. That is the opposite of anemic.

Skip a getter whose source is `ρ`. Or, if the flat case is still worth reporting, only report when no descendant formation refers to the name, since that is exactly the case where the replacement is a plain `^`.

**Comment by 0crat** [comment]
@maxonfjvipon Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+1001](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
